fdbserver/TLogServer.actor.cpp

```diff
@@ -1758,7 +1759,17 @@ Future<Void> tLogPeekMessages(PromiseType replyPromise,
 
 	if (SERVER_KNOBS->ENABLE_VERSION_VECTOR && reqBegin > logData->persistentDataDurableVersion && !reqOnlySpilled &&
 	    reqTag.locality >= 0 && !reqReturnIfBlocked) {
-           wait(waitForMessagesForTag(logData, reqTag, reqBegin, SERVER_KNOBS->BLOCKING_PEEK_TIMEOUT));
+		state double startTime = now();
+		wait(waitForMessagesForTag(logData, reqTag, reqBegin, SERVER_KNOBS->BLOCKING_PEEK_TIMEOUT));
+		double latency = now() - startTime;
+		if (logData->blockingPeekLatencies.find(reqTag) == logData->blockingPeekLatencies.end()) {
+			UID ssID = nondeterministicRandom()->randomUniqueID();
+			std::string s = "BlockingPeekLatencies " + reqTag.toString();
+			logData->blockingPeekLatencies.try_emplace(
+			    reqTag, s, ssID, SERVER_KNOBS->LATENCY_METRICS_LOGGING_INTERVAL, SERVER_KNOBS->LATENCY_SAMPLE_SIZE);
+		}
+		LatencySample& sample = logData->blockingPeekLatencies.at(reqTag);
+		sample.addMeasurement(latency);
```

Comment:
Don't you need to make changes to Status.actor.cpp to expose these metrics to the user? How did you capture these metrics in your experiments without making that change? Is there some other way to collect these metrics?

Reply:
The `LatencySample` [constructor](https://github.com/apple/foundationdb/blob/52a2f56fe25d7a4bb376ecc1648d4253980c0839/fdbrpc/Stats.h#L228-L232) launches a coroutine that periodically writes the output to the logs.